esp/esdllib/esdl_transformer2.cpp

```diff
@@ -1515,6 +1515,18 @@ int Esdl2Transformer::process(IEspContext &ctx, EsdlProcessMode mode, const char
     if (!mi)
         throw MakeStringException(-1, "ESDL - method '%s::%s'not found", service, method);
 
+    // check the auth_feature
+    if (mode==EsdlRequestMode)
+    {
+        const char* auth_feature = mi->queryMetaData("auth_feature");
+        if (auth_feature && *auth_feature)
+        {
+            SecAccessFlags access;
+            if (!ctx.authorizeFeature(auth_feature,access) || access!=SecAccess_Full)
```

Comment:
the auth_feature level should be configurable, here it's always assumed "FULL", even if SCApps will always set to full.
auth_feature("featurename:level");